python/gstaichi/lang/kernel_impl.py

```diff
@@ -997,6 +1016,8 @@ def recursive_set_args(needed_arg_type: Type, provided_arg_type: Type, v: Any, i
                 set_arg_sparse_matrix_builder(indices, v)
                 return 1
             if dataclasses.is_dataclass(needed_arg_type):
+                if provided_arg_type != needed_arg_type:
+                    print("needed", needed_arg_type, "!= provided", provided_arg_type)
                 assert provided_arg_type == needed_arg_type
```

Comment:
How about raising `GsTaichiRuntimeTypeError` in the above if statement and remove this line?